src/CoreWCF.WebHttp/src/CoreWCF/OpenApi/Middleware/OpenApiMiddleware.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace CoreWCF.OpenApi.Middleware
+{
+    internal sealed class OpenApiMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly OpenApiDocumentProvider _openApiDocumentProvider;
+        private readonly Dictionary<string, byte[]> _swaggerUi;
+
+        public OpenApiMiddleware(
+            RequestDelegate next,
+            OpenApiDocumentProvider openApiDocumentProvider)
+        {
+            _next = next;
+            _openApiDocumentProvider = openApiDocumentProvider;
+            _swaggerUi = new Dictionary<string, byte[]>
+            {
+                { "/swagger/favicon-16x16.png", ReadResource("favicon-16x16.png") },
+                { "/swagger/favicon-32x32.png", ReadResource("favicon-32x32.png") },
+                { "/swagger/index.html", ReadResource("index.html") },
+                { "/swagger/swagger-ui-bundle.js", ReadResource("swagger-ui-bundle.js") },
+                { "/swagger/swagger-ui-standalone-preset.js", ReadResource("swagger-ui-standalone-preset.js") },
+                { "/swagger/swagger-ui.css", ReadResource("swagger-ui.css") },
+            };
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            string requestPath = context.Request.Path.ToString();
+            if (!requestPath.StartsWith("/"))
+            {
+                requestPath = "/" + requestPath;
+            }
+
+            // Needs to happen after contracts have been added to document provider.
+            if (OpenApiDocumentationEnabled() && requestPath == "/swagger/swagger.json")
+            {
+                await WriteResponse(context, requestPath, Encoding.UTF8.GetBytes(_openApiDocumentProvider.Document));
+            }
```

Comment:
Should this be added to the dictionary instead of being special cased? If there's an issue of initialization order, maybe have OpenApiDocumentationEnabled lazily add the swagger json file?